cstar/base/component.py

```diff
@@ -87,43 +90,34 @@ def __str__(self) -> str:
 
         NID = len(self.input_datasets)
 
-        base_str += f"\n{NAC} AdditionalCode repositories (query using Component.additional_code)"
         base_str += (
-            f"\n{NID} InputDataset objects (query using Component.input_datasets"
+            f"\n{NAC} AdditionalCode instances (query using Component.additional_code)"
         )
-
-        # Discretisation
-        disc_str = ""
-        if hasattr(self, "time_step") and self.time_step is not None:
-            disc_str += "\ntime_step: " + str(self.time_step) + "s"
-        if hasattr(self, "n_procs_x") and self.n_procs_x is not None:
-            disc_str += (
-                "\nn_procs_x: "
-                + str(self.n_procs_x)
-                + " (Number of x-direction processors)"
-            )
-        if hasattr(self, "n_procs_y") and self.n_procs_y is not None:
-            disc_str += (
-                "\nn_procs_y:"
-                + str(self.n_procs_y)
-                + " (Number of y-direction processors)"
-            )
-        if hasattr(self, "n_levels") and self.n_levels is not None:
-            disc_str += "\nn_levels:" + str(self.n_levels)
-        if hasattr(self, "nx") and self.nx is not None:
-            disc_str += "\nnx:" + str(self.nx)
-        if hasattr(self, "ny") and self.ny is not None:
-            disc_str += "\nny:" + str(self.ny)
+        base_str += f"\n{NID} Input datasets (query using Component.input_datasets)"
+        if hasattr(self, "discretization") and self.discretization is not None:
+            base_str += "\n\nDiscretization:\n"
+            base_str += self.discretization.__str__()
         if hasattr(self, "exe_path") and self.exe_path is not None:
-            disc_str += "\n\nIs compiled: True"
-            disc_str += "\n exe_path: " + self.exe_path
-        if len(disc_str) > 0:
-            disc_str = "\n\nDiscretization info:" + disc_str
-        base_str += disc_str
+            base_str += "\n\nIs compiled: True"
+            base_str += "\n exe_path: " + self.exe_path
         return base_str
 
     def __repr__(self) -> str:
-        return self.__str__()
+        repr_str = f"{self.__class__.__name__}("
+        repr_str += f"\nbase_model = <{self.base_model.__class__.__name__} instance>, "
+        if self.additional_code is not None:
+            repr_str += f"\nadditional_code = <{self.additional_code.__class__.__name__} instance>, "
+        else:
+            repr_str += "\n additional_code = None"
```

Comment:
Isn't there some redundant here with the other classes? i.e. you could be calling `repr(additional_code)` here.

Reply:
As the class structure is so hierarchical I tried to avoid a cascading printout of hundreds of lines by just calling reprs within reprs all the way down. This is a change from #74 however, would be good to discuss strs/reprs over there as that PR in particular was supposed to be a first stab at something that could become a discussion.